### PR TITLE
add new methods to captum and tf-explain funcs

### PR DESCRIPTION
**Pull request by dilyabareeva**
### Description
- Addressing the issue [153](https://github.com/understandable-machine-intelligence-lab/Quantus/issues/153).
- Extend the list of available explanation methods in Quantus
- Rename existing attribution method strings to match original function names (ex. Gradient->VanillaGradients as per tf-explain, GradCAM->LayerGradCam as per captum)

### Implemented changes
  - [ ] New captum methods: 
DeepLift 
DeepLiftShap 
GuidedGradCam 
Deconvolution 
FeaturePermutation
Lime
KernelShap
LRP
LayerConductance
LayerActivation
InternalInfluence
LayerGradientXActivation
  - [ ] New tf-explain methods:
SmoothGrad
  - [ ] Black formatting
  - [ ] Add to constants.py (separation captum, tf-explain)
  - [ ] Tests for new methods

### CODE REVIEW

**Comment by annahedstroem** [comment]
Hi @dilyabareeva! Looks great. If we have re-named existing methods like GradCAM to LayerGradCAM can we let the user still use the GradCAM but automatically direct them to LayerGradCAm together with a deprecation print message? Otherwise, users' code will break which is unexpected in a minor version update. I know we have one earlier deprecation warning in warn_func, check there!

File: quantus/functions/explanation_func.py [hunk 1]
```diff
@@ -352,42 +387,37 @@ def f_reduce_axes(a):
 
     explanation: torch.Tensor = torch.zeros_like(inputs)
 
-    if method == "GradientShap".lower():
+    if method in ["GradientShap", "IntegratedGradients", "DeepLift", "DeepLiftShap"]:
```
Comment (annahedstroem) [inline quantus/functions/explanation_func.py:390]: Great that you group them here!

File: quantus/functions/explanation_func.py [hunk 2]
```diff
@@ -352,42 +387,37 @@ def f_reduce_axes(a):
...
-                baselines=kwargs.get("baseline", torch.zeros_like(inputs)),
-                n_steps=10,
-                method="riemann_trapezoid",
-            )
-        )
-
-    elif method == "InputXGradient".lower():
-        explanation = f_reduce_axes(
-            InputXGradient(model).attribute(inputs=inputs, target=targets)
+            attr_func(model).attribute(inputs=inputs, target=targets)
```
Comment (annahedstroem) [inline quantus/functions/explanation_func.py:412]: What if the user has kwargs for the explanation method? Where does it go? Not in attr_func?

Comment (annahedstroem) [reply quantus/functions/explanation_func.py:412]: @dilyabareeva check this when you can :)

**Review by annahedstroem — COMMENTED** [review]
Ok, ready for merge.

**Comment by annahedstroem** [comment]
@dilyabareeva can you resolve the merge conflicts?
